Library/Homebrew/compat/macos.rb

```diff
@@ -2,97 +2,170 @@
+
+      def lion_or_newer?
+        opoo "Calling MacOS.lion_or_newer? is deprecated!"
+        puts "Use 'MacOS.version >= :lion' instead."
+        version >= :lion
+      end
+
+      def mountain_lion?
+        opoo "Calling MacOS.mountain_lion? is deprecated!"
+        puts "Use 'MacOS.version >= :mountain_lion' instead."
+        version >= :mountain_lion
+      end
+
+      def mountain_lion_or_newer?
+        opoo "Calling MacOS.mountain_lion_or_newer? is deprecated!"
+        puts "Use 'MacOS.version >= :mountain_lion' instead."
+        version >= :mountain_lion
+      end
+
+      def macports_or_fink_installed?
+        opoo "Calling MacOS.macports_or_fink_installed? is deprecated!"
+        puts "Use !MacOS.macports_or_fink.empty? instead."
+        !macports_or_fink.empty?
+      end
+
+      def locate(tool)
+        opoo "Calling MacOS.locate is deprecated!"
+        puts "Use DevelopmentTools.locate instead."
+        DevelopmentTools.locate(tool)
+      end
+
+      def default_cc
+        opoo "Calling MacOS.default_cc is deprecated!"
+        puts "Use DevelopmentTools.default_cc instead."
+        DevelopmentTools.default_cc
+      end
+
+      def default_compiler
+        opoo "Calling MacOS.default_compiler is deprecated!"
+        puts "Use DevelopmentTools.default_compiler instead."
+        DevelopmentTools.default_compiler
+      end
+
+      def gcc_40_build_version
+        opoo "Calling MacOS.gcc_40_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_40_build_version instead."
+        DevelopmentTools.gcc_40_build_version
+      end
+
+      def gcc_4_0_build_version
+        opoo "Calling MacOS.gcc_4_0_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_40_build_version instead."
+        DevelopmentTools.gcc_40_build_version
+      end
+
+      def gcc_42_build_version
+        opoo "Calling MacOS.gcc_42_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_42_build_version instead."
+        DevelopmentTools.gcc_42_build_version
+      end
+
+      def gcc_build_version
+        opoo "Calling MacOS.gcc_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_42_build_version instead."
+        DevelopmentTools.gcc_42_build_version
+      end
+
+      def llvm_build_version
+        opoo "Calling MacOS.llvm_build_version is deprecated!"
+        puts "There is no replacement."
+      end
+
+      def clang_version
+        opoo "Calling MacOS.clang_version is deprecated!"
+        puts "Use DevelopmentTools.clang_version instead."
+        DevelopmentTools.clang_version
+      end
+
+      def clang_build_version
+        opoo "Calling MacOS.clang_build_version is deprecated!"
+        puts "Use DevelopmentTools.clang_build_version instead."
+        DevelopmentTools.clang_build_version
+      end
+
+      # We won't change the name because of backward compatibility.
+      # So disable rubocop here.
+      # rubocop:disable Style/PredicateName
```

Comment:
Again, I'm intentionally disabling for the whole file. Ideally we'd disable this for all of `compat/`